backend/app/services/implementations/meal_request_service.py

```diff
@@ -94,6 +96,59 @@ def update_meal_request(
         original_meal_request.save()
         return meal_request_dto
 
+    def commit_to_meal_request(
+        self,
+        donor_id: str,
+        meal_request_ids: [str],
+        food_description: str,
+        additional_info: str,
+    ) -> [MealRequestDTO]:
+        try:
+            donor = User.objects(id=donor_id).first()
+            if not donor:
+                raise Exception(f'user "{donor_id}" not found')
+            # The user committing to the meal request must have the "Donor" role
+            if donor.info.role != UserInfoRole.DONOR.value:
+                raise Exception(f'user "{donor_id}" is not a donor')
+
+            if len(meal_request_ids) == 0:
+                raise Exception("no meal requests to commit to")
+
+            meal_request_dtos = []
+            for meal_request_id in meal_request_ids:
+                meal_request = MealRequest.objects(id=meal_request_id).first()
+                if not meal_request:
+                    raise Exception(f'meal request "{meal_request_id}" not found')
+                # The meal request must be in the "Open" status
+                if meal_request.status != MealStatus.OPEN.value:
+                    raise Exception(
+                        f'meal request "{meal_request_id}" is not open for commitment'
+                    )
+
+                meal_request.donation_info = DonationInfo(
+                    donor=donor,
+                    commitment_date=datetime.utcnow(),
+                    food_description=food_description,
+                    additional_info=additional_info,
+                )
+
+                # Change the meal request's status to "Fulfilled"
```

Comment:
I was thinking, should we define an additional meal request status called `Pending`, for meal requests that have been committed to but not donated yet? To me, `Fulfilled` implies that the meal request has been donated to the ASP and the request is now closed.

Reply:
Hey William, agree with this! 

Originally it was done this way since after a meal donor committed to a request, we never got anymore confirmation for that meal request (since we don't require the user to come back on the platform and mark that "the delivery is done"), so from the platforms perspective that meal request is done, and there is no more chances to change it from pending to fulfilled. 

However, since we now plan on adding reminder and thank you emails, we're going to have to setup an asynchronous job that gets activated after and before the donation date, and thus we can have that update the status, so lets add the more refined status!

Since the change is small and will effect this PR, could you make the change in this PR? 

Lets call it `UPCOMING` since in the admin UI these meal requests (that have a meal donor but have not happened yet) are called that.

Reply:
Makes sense, done :)